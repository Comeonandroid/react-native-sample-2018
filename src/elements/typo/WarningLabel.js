import React from 'react'
import styled from 'styled-components'
import { PropTypes } from 'prop-types'

import CustomIcon from '../../utils/icon'
import { Text, View } from '../index'


const WarningLabel = ({ children, ...rest }) => (
  <Wrapper>
    <Icon name='warning' size={25} />
    <Element {...rest}>
      {children}
    </Element>
  </Wrapper>
)

WarningLabel.propTypes = {
  children: PropTypes.node
}

const Element = styled(Text) `
  flex: 1;
  flex-wrap: wrap;
  margin: 5px 0px;
  font-family: Open Sans;
  font-weight: 600;
  font-size: 11px;
`

const Wrapper = styled(View) `
  flex-wrap: wrap;
  flex-direction: row;
  align-items: center;
`

const Icon = styled(CustomIcon) `
  margin-right: 15px;
  color: ${ ({ theme }) => theme.colors.primary }
`

export default WarningLabel