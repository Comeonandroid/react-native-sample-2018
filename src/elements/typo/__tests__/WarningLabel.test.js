import 'react-native'
import React from 'react'
import { itBehavesLike } from '../../../specs/shared_examples/index'

import WarningLabel from '../WarningLabel'

describe('<WarningLabel />', () => {
  itBehavesLike('aThemedComponent', { subject: <WarningLabel>Test</WarningLabel> })
})
