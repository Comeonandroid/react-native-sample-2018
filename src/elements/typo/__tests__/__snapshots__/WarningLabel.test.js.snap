// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<WarningLabel /> dark: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        },
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 25,
        },
        Array [
          Object {
            "color": "#fcfcfc",
          },
          Array [
            Object {
              "color": "#faa632",
              "marginRight": 15,
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#fcfcfc",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "wrap",
            "fontFamily": "Open Sans",
            "fontSize": 11,
            "fontWeight": "600",
            "marginBottom": 5,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 5,
          },
          undefined,
        ],
      ]
    }
  >
    Test
  </Text>
</View>
`;

exports[`<WarningLabel /> light: renders correctly 1`] = `
<View
  style={
    Array [
      Object {},
      Array [
        Object {
          "alignItems": "center",
          "flexDirection": "row",
          "flexWrap": "wrap",
        },
        undefined,
      ],
    ]
  }
>
  <Text
    accessible={true}
    allowFontScaling={false}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": undefined,
          "fontSize": 25,
        },
        Array [
          Object {
            "color": "#4b4c4d",
          },
          Array [
            Object {
              "color": "#faa632",
              "marginRight": 15,
            },
            undefined,
          ],
        ],
        Object {
          "fontFamily": "icomoon",
          "fontStyle": "normal",
          "fontWeight": "normal",
        },
      ]
    }
  >
    
  </Text>
  <Text
    accessible={true}
    allowFontScaling={true}
    ellipsizeMode="tail"
    style={
      Array [
        Object {
          "color": "#000000",
          "fontFamily": "Open Sans",
        },
        Array [
          Object {
            "flexBasis": 0,
            "flexGrow": 1,
            "flexShrink": 1,
            "flexWrap": "wrap",
            "fontFamily": "Open Sans",
            "fontSize": 11,
            "fontWeight": "600",
            "marginBottom": 5,
            "marginLeft": 0,
            "marginRight": 0,
            "marginTop": 5,
          },
          undefined,
        ],
      ]
    }
  >
    Test
  </Text>
</View>
`;
